lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -674,83 +843,446 @@ export default function getAPIImplementation(): APIInterface {
+                                password,
+                                shouldAttemptAccountLinkingIfAllowed: true,
+                                userContext,
+                            });
+                            if (signUpResponse.status === "EMAIL_ALREADY_EXISTS_ERROR") {
+                                return signUpResponse;
+                            }
+
+                            session = await Session.createNewSession(
+                                options.req,
+                                options.res,
+                                tenantId,
+                                signUpResponse.recipeUserId,
+                                {},
+                                {},
+                                userContext
+                            );
+
+                            await mfaInstance.recipeInterfaceImpl.markFactorAsCompleteInSession({
+                                session: session,
+                                factorId: "emailpassword",
+                                userContext,
+                            });
+
+                            return {
+                                status: "OK",
+                                session,
+                                user: signUpResponse.user,
+                            };
+                        } else {
+                            // secondary factors
+                            let sessionUser = await getUser(session.getUserId(), userContext);
+                            if (sessionUser === undefined) {
+                                throw new SessionError({
+                                    type: SessionError.UNAUTHORISED,
+                                    message: "Session user not found",
+                                });
+                            }
+
+                            // MFA is enabled
+                            // Active session / secondary factor
+                            // Sign Up / Factor setup
+
+                            checkFactorUserAccountInfoForVerification(sessionUser, { email });
+
+                            await MultiFactorAuth.assertAllowedToSetupFactorElseThrowInvalidClaimError(
+                                session,
+                                "emailpassword",
+                                userContext
+                            );
+
+                            await checkIfFactorUserBeingCreatedCanBeLinkedWithSessionUser(
+                                tenantId,
+                                sessionUser,
+                                { email },
+                                userContext
+                            );
+
+                            let signUpResponse = await options.recipeImplementation.signUp({
+                                tenantId,
+                                email,
+                                password,
+                                shouldAttemptAccountLinkingIfAllowed: false,
```

Comment:
wherever we are calling signUp with false for shouldAttemptAccountLinkingIfAllowed, we should call createNewReceipeUser instead.

Reply:
Done